### PR TITLE
register & install

### PR DESCRIPTION
**Pull request by MHBauer**
got the endpoint to actually show up. Example into docs.

@arschles  not trying to snipe any work from #172 or #163. let's get together if we're conflicting.

### CODE REVIEW

**Comment by arschles** [comment]
@MHBauer no worries at all!

Looks like this subsumes #163, so I've closed that PR

#172 starts to add `rest.Storage`s for each resource but I haven't gotten far. Let me know if you're planning on doing that soon and I'll close that one too.

**Comment by MHBauer** [comment]
It's all one big mish-mash in my environment.

I started over with another example. This was the simplest thing that worked. It shows forward progress, so I PR'd it.

**Comment by duglin** [comment]
rebase needed

**Comment by pmorie** [comment]
:100: LGTM

**Comment by duglin** [comment]
LGTM once CI is happy

**Comment by MHBauer** [comment]
changes for lint.

**Comment by duglin** [comment]
still LGTM

**Comment by pmorie** [comment]
@MHBauer rebase needed -- let's get this in today.

File: pkg/apis/servicecatalog/install/install.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import (
+	"k8s.io/kubernetes/pkg/apimachinery/announced"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1alpha1"
+)
+
+func init() {
+	if err := announced.NewGroupMetaFactory(
```
Comment (pmorie) [inline pkg/apis/servicecatalog/install/install.go]: @deads2k could you answer these questions, if you have the information?

File: pkg/apis/servicecatalog/install/install.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+		&announced.GroupMetaFactoryArgs{
+			GroupName:              servicecatalog.GroupName,
+			VersionPreferenceOrder: []string{v1alpha1.SchemeGroupVersion.Version},
+			ImportPrefix:           "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog",
+			// TODO: what does this do?
```
Comment (deads2k) [inline pkg/apis/servicecatalog/install/install.go]: there's some reflect in this path that tries to pull out type names and needs to know which packages it came from to properly isolate what you're adding.

File: pkg/apis/servicecatalog/install/install.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+			RootScopedKinds: nil, // nil is allowed
+			// TODO: Do we have 'internal objects'? What is an 'internal object'?
```
Comment (deads2k) [inline pkg/apis/servicecatalog/install/install.go]: yes.  it's a hub version for a normal serialize at the edges, deserialize to hub, convert back out to any edge pattern.

File: pkg/apis/servicecatalog/install/install.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+			// mhb: ? broker/catalog/service/instance are our 'internal objects' ?
```
Comment (deads2k) [inline pkg/apis/servicecatalog/install/install.go]: your path looks weird.  normally its `/apis/<group>/types.go`

File: pkg/apis/servicecatalog/install/install.go [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+			AddInternalObjectsToScheme: servicecatalog.AddToScheme, // nil if there are no 'internal objects'
+		},
+		// TODO what does this do? Is it necessary?
```
Comment (deads2k) [inline pkg/apis/servicecatalog/install/install.go]: This I don't know.  Part of @lavalamp's refactoring at some point.

File: pkg/apis/servicecatalog/v1alpha1/register.go [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+var (
+	schemeBuilder = runtime.NewSchemeBuilder(addKnownTypes)
+	// AddToScheme is exposed for API installation
+	AddToScheme = schemeBuilder.AddToScheme
+)
+
+func addKnownTypes(scheme *runtime.Scheme) error {
+	scheme.AddKnownTypes(SchemeGroupVersion,
+		// TODO are all of these needed? What do they do?
```
Comment (arschles) [inline pkg/apis/servicecatalog/v1alpha1/register.go]: create an issue to get this answered?

File: pkg/apiserver/apiserver.go [hunk 7]
```diff
@@ -58,5 +61,22 @@ func (c CompletedConfig) New() (*ServiceCatalogAPIServer, error) {
...
 	}
 
+	glog.Infoln("make and install the apis")
+
+	apiGroupInfo := genericapiserver.NewDefaultAPIGroupInfo(servicecatalog.GroupName)
+	// giving it v1alpha1 version
+	apiGroupInfo.GroupMeta.GroupVersion = v1alpha1.SchemeGroupVersion
+
+	// TODO make storage work
+	//v1alpha1storage := map[string]rest.Storage{}
```
Comment (arschles) [inline pkg/apiserver/apiserver.go]: work being done in https://github.com/kubernetes-incubator/service-catalog/pull/172

**Pull request merged**